tests/benches/noncompressed/src/main.rs

```diff
@@ -0,0 +1,108 @@
+use byteorder::WriteBytesExt;
+use std::fs::File;
+use std::io::Write;
+
+fn write_png_sig(w: &mut impl Write) {
+    const SIG: [u8; 8] = [137, 80, 78, 71, 13, 10, 26, 10];
+    w.write_all(&SIG).unwrap();
+}
+
+fn write_chunk(w: &mut impl Write, chunk_type: &[u8], data: &[u8]) {
+    let crc = {
+        let input = chunk_type
+            .iter()
+            .copied()
+            .chain(data.iter().copied())
+            .collect::<Vec<_>>();
+        crc32fast::hash(input.as_slice())
+    };
+    w.write_u32::<byteorder::BigEndian>(data.len() as u32)
+        .unwrap();
+    w.write_all(chunk_type).unwrap();
+    w.write_all(data).unwrap();
+    w.write_u32::<byteorder::BigEndian>(crc).unwrap();
+}
+
+fn write_ihdr(w: &mut impl Write, width: u32) {
+    let mut data = Vec::new();
+    data.write_u32::<byteorder::BigEndian>(width).unwrap();
+    data.write_u32::<byteorder::BigEndian>(width).unwrap(); // height
+    data.write_u8(8).unwrap(); // bit depth = always 8-bits per channel
+    data.write_u8(6).unwrap(); // color type = color + alpha
+    data.write_u8(0).unwrap(); // compression method (0 is the only allowed value)
+    data.write_u8(0).unwrap(); // filter method (0 is the only allowed value)
+    data.write_u8(0).unwrap(); // interlace method = no interlacing
+                               //
+    write_chunk(w, b"IHDR", &data);
+}
+
+fn write_noncompressed_deflate_block(w: &mut impl Write, data: &[u8], last: bool) {
+    if last {
+        w.write_u8(1).unwrap();
+    } else {
+        w.write_u8(0).unwrap();
+    }
+
+    assert!(data.len() < 65535);
+    let len = data.len() as u16;
+    w.write_u16::<byteorder::LittleEndian>(len).unwrap();
+    w.write_u16::<byteorder::LittleEndian>(!len).unwrap();
+
+    w.write_all(data).unwrap();
+}
+
+fn write_idat(w: &mut impl Write, width: u32) {
+    // Generate arbitrary test pixels.
+    let image_pixels = {
+        let mut row = Vec::new();
+        row.write_u8(0).unwrap(); // filter = no filter
+
+        let row_pixels = (0..width)
+            .map(|i| {
+                let color: u8 = (i * 255 / width) as u8;
+                let alpha: u8 = 0xff;
+                [color, 255 - color, color / 2, alpha]
+            })
+            .flatten();
+        row.extend(row_pixels);
+
+        std::iter::repeat(row)
+            .take(width as usize)
+            .flatten()
+            .collect::<Vec<_>>()
+    };
+
+    let mut zlib_data = Vec::new();
+    zlib_data.write_u8(0x78).unwrap(); // CM + CINFO
+    zlib_data.write_u8(0x01).unwrap(); // FLG
+
+    let mut chunks = image_pixels.as_slice().chunks_exact(8192);
```

Comment:
Is there a particular reason to prefer 8KB blocks rather than the max size? If not, fdeflate has a [`StoredOnlyCompressor`](https://docs.rs/fdeflate/latest/fdeflate/struct.StoredOnlyCompressor.html) that could replace this manual deflate stream construction

Reply:
Thanks!  TIL :-).  I should have checked `fdeflate` before writing this code (I've struggled for quite a bit to produce a working zlib stream).  I'll make the switch in a few minutes.

Reply:
Done